src/js/pages/GranthIndex/components/GranthIndex.tsx

```diff
@@ -1,38 +1,150 @@
-import React, { useEffect, MouseEvent } from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
 import { Link } from 'react-router-dom';
 import { Location } from 'history';
 
 import BreadCrumb from '@/components/Breadcrumb';
 import Fetch from '@/components/Fetch';
-import { raagIndices } from '../constants/raag-indices';
+import { raagIndices, IRaagIndex } from '../constants/raag-indices';
 import { toAngURL } from '../../../util';
 import { pageView } from '../../../util/analytics';
 import { AmritKeertanIndex } from './AmritKeertanIndex';
-import { sanitizeHash, scrollToHash } from '../util';
+import { sanitizeHash, scrollToHash, desanitzeHash } from '../util';
 import { TEXTS } from '../../../constants';
 
 interface IGranthIndexProps {
-  location: Location
+  location: Location;
 }
 
+const GRANTH_NAMES = raagIndices.map(({ name }) => name);
+
 export const GranthIndex: React.FC<IGranthIndexProps> = ({ location }) => {
   const { hash } = location;
 
+  const [selection, setSelection] = useState<string[]>(() =>
+    hash ? desanitzeHash(hash) : [GRANTH_NAMES[0]]
+  );
+
+  // Helps check if a list item is selected to show subSets and description
+  const isSelected = useCallback(
+    (name: string) => {
+      if (selection.includes(name)) {
+        return true;
+      }
+    },
+    [selection]
+  );
+
+  // Called onClick to show description and subSets
+  const changeSelection = useCallback(
+    (path: string[]) => {
+      if (path.join() == selection.join()) {
+        path.pop();
+      }
+      setSelection(path);
+    },
+    [selection]
+  );
+
   useEffect(() => {
     pageView('/index');
     // making sure the element is rendered on DOM before we get to that
     setTimeout(() => scrollToHash(hash), 800);
-  }, [])
+  }, []);
+
+  // Recursive function that keeps churning lists until no subSets are left
+  // path is a list that helps keep the parents open when a child is clicked upon
+  const renderList = useCallback(
+    (
+      subSets: IRaagIndex[],
+      source: IRaagIndex['source'] = 'G',
+      path: string[] = []
+    ) => {
+      return subSets.map(
+        (
+          {
+            name,
+            pages: [from, to],
+            source: localSource,
+            description,
+            subSets,
+            // Give a default value if highlight is undefined
+            highlight = 0,
+          },
+          i
+        ) => {
+          if (localSource == null) {
+            localSource = source;
+          } else {
+            source = localSource;
+          }
```

Comment:
@yudhv not sure I am following here properly. Couldn't see the source variable being used.

Reply:
Thank you. I removed this chunk and tested the source functionality.